kotlin/codegen/build.gradle.kts

```diff
@@ -65,9 +65,11 @@ dependencies {
     exclude(group = "org.jetbrains.kotlin")
     exclude(group = "com.squareup", module = "kotlinpoet")
   }
+  api(Dependencies.KotlinPoet.elementsClassInspector)
   shade(Dependencies.KotlinPoet.elementsClassInspector) {
     exclude(group = "org.jetbrains.kotlin")
     exclude(group = "com.squareup", module = "kotlinpoet")
+    exclude(group = "com.google.guava")
```

Comment:
what were these for?

Reply:
I don't understand `shadowJar` well, but it seems that `guava` has been added to the dependencies of `kotlinpoet-classinspector-elements` from 1.7.x, which broke the jar file of the output of the `shadowJar` task and caused `kapt` to fail.
Therefore, `guava` was excluded to make the build successful.

Reply:
That makes me uncomfortable as kotlinpoet uses guava APIs then, removing it likely breaks the artifact. Can you be more specific about what issues you encountered?

Reply:
Before I excluded `guava` (cdc51e9), the `:kotlin:tests:compileTestKotlin` task had `Unresolved reference: GeneratedAdaptersTest_MultipleProperti esShareAdapterJsonAdapter` error.
In this case, the `:kotlin:tests:kaptTestKotlin` task did not give an error and I could not find a solution.

Next I thought there might be a problem with `shadowJar`.

Here are the contents of the `jar` file that is the artifact of the `:kotlin:codegen:shadowJar` task before I excluded `guava` (cdc51e9).

KotlinPoet 1.6 | KotlinPoet 1.7
-- | --
![image](https://user-images.githubusercontent.com/5885032/108499023-0b778880-72f1-11eb-839f-8616e3c3e636.png) | ![image](https://user-images.githubusercontent.com/5885032/108499076-1d592b80-72f1-11eb-9083-c6a46c72aba7.png)

When I saw this, I thought that `guava` might have broken `shadowJar`, so I excluded `guava` from the `shade` dependency. Then `./gradlew build` got successful.

I don't know why moshi using `shadowJar`, should I write the `rerocate` configuration for `guava` in `shadowJar` block?

Reply:
Thanks for the detailed writeup. In this case, I think it's ok because we're telling it just to exclude guava from shading but not from the actual API